packages/client/test/stream.test.ts

```diff
@@ -66,6 +76,70 @@ describe("Stream", async () => {
     }
   });
 
+  test("Stream with lot's of events", async () => {
+    const testCreateItem = {
+      updateItem: {
+        eventId: Buffer.from([1, 2, 3, 4]),
+        price: "1000",
+        metadata: Buffer.from([1, 2, 3, 4]),
+      },
+    };
+
+    const signedMessage = await signMessage(testCreateItem);
+    const events = [];
+
+    for (let index = 0; index < 50; index++) {
+      events.push(signedMessage);
+    }
+    const pushEvent = {
+      requestId: Uint8Array.from([1, 2, 3, 4]),
+      events,
+    };
+    const client = new MockClient();
+    const stream = new ReadableEventStream(client);
+    stream.enqueue(pushEvent);
+    let count = 0;
+    for await (const evt of stream.stream) {
+      count++;
+      if (count === pushEvent.events.length) break;
+    }
+  });
+
+  test("Stream filled with test vectors", async () => {
+    const events = [];
+    for (let index = 0; index < testVectors.events.length; index++) {
+      const evt = testVectors.events[index];
+      events.push({
+        signature: hexToBytes(("0x" + evt.signature) as `0x${string}`),
+        event: {
+          type_url: "type.googleapis.com/market.mass.ShopEvent",
+          value: hexToBytes(("0x" + evt.encoded) as `0x${string}`),
+        },
+      });
+    }
+
+    const pushReq = new schema.EventPushRequest({
+      requestId: randomBytes(16),
+      events,
+    });
+
+    const client = new MockClient();
+    const stream = new ReadableEventStream(client);
+    stream.enqueue(pushReq);
+    // TODO: we need a way to close the stream once all requests have been pushed
+    let count = 0;
+    for await (const evt of stream.stream) {
+      count++;
+      // this might never happen if events are omitted by ReadableEventStream
+      // TODO: this might also exit to early if events would be duplicated...
+      if (count === pushReq.events.length) break;
+    }
+    expect(count).toEqual(testVectors.events.length);
+  });
+
+  // TODO: should add version of the test above that creates random chunking of the test vectors
+  // and assert that they are all returned in order as individual events
```

Comment:
@jchoi330 you might want to try taking a stab at this but i think it can wait until a follow-up PR